### PR TITLE
[KYUUBI #703][FOLLOWUP][DEPS][TEST] Reduce Hudi dependencies

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

HBase deps is optional for creating hudi table and query, but required for writing data to hudi table, as currently we only test create table and query for data lake formats, it's reasonable to remove the hbase deps to reduce the deps pulled from hudi.

This PR also change all data lake test suites in spark engine from `hive` catalog to `in-memory` catalog.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
@simon824 seems that HUDI test suites passed without any HBase depenedencies

File: kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala [hunk 1]
```diff
@@ -42,5 +42,6 @@ trait HudiSuiteMixin extends DataLakeSuiteMixin {
     "spark.sql.defaultCatalog" -> catalog,
     "spark.sql.extensions" -> "org.apache.spark.sql.hudi.HoodieSparkSessionExtension",
     "spark.serializer" -> "org.apache.spark.serializer.KryoSerializer",
+    "hoodie.bootstrap.index.class" -> "org.apache.hudi.common.bootstrap.index.HFileBootstrapIndex",
```
Comment (pan3793) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala:45]: Actually, `HFileBootstrapIndex` is the default value, it's not matter with/without this conf I think.

**Comment by simon824** [comment]
> @simon824 seems that HUDI test suites passed without any HBase depenedencies

Create operation is working fine,but insert /update operation will cause exception, does ut need add  insert /update operation ?

File: kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala (hunk 1 above)
Comment (simon824) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala:45]: Yes, HFileBootstrapIndex is the default value, if you want reduce hbase dep, this config should be 'org.apache.hudi.common.bootstrap.index.NoOpBootstrapIndex'.

**Comment by pan3793** [comment]
The current ut for data lake formats only contain create and query, I think it's OK to remove the hbase deps now, and maybe hudi will have improved the deps management when we add the write test cases in the future.

**Comment by simon824** [comment]
> The current ut for data lake formats only contain create and query, I think it's OK to remove the hbase deps now, and maybe hudi will have improved the deps management when we add the write test cases in the future.

OK

File: kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala (hunk 1 above)
Comment (pan3793) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala:45]: Yeah, just change it back to default for test. BTW, is it the right approach to overwrite hudi conf?

Comment (simon824) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala]: Overwriting the config when creating the table

https://hudi.apache.org/docs/quick-start-guide
image

File: kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala [hunk 2]
```diff
@@ -38,7 +38,6 @@ trait HudiSuiteMixin extends DataLakeSuiteMixin {
   }
 
   override protected def extraConfigs = Map(
-    "spark.sql.catalogImplementation" -> "in-memory",
```
Comment (simon824) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala:41]: this config is necessary

Comment (pan3793) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala:41]: Yeah, I see, just move it to `WithSparkSQLEngine`. 
BTW, why hudi doesn't work with `hive` catalog? `in-memory` should only used for test, I think most users would use hive for production.

File: kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala (hunk 1 above)
Comment (pan3793) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala]: Thanks, we need refactor `DataLakeSuiteMixin` and expose create table options to subclass in future.

**Comment by pan3793** [comment]
cc @yanghua @yaooqinn

File: kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala (hunk 2 above)
Comment (simon824) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala]: It work well if hive metastore is base on mysql  and will cause Concurrent Exception  if base on derby. 
Derby is the default config value, and ut is base on it .

Comment (pan3793) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala]: How does Hudi community do such a Hive integration test?

Comment (simon824) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/HudiSuiteMixin.scala]:
```
spark.sql("...")
```
It works well with derby.

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master for v1.4.0 and unblock #1029

File: pom.xml [hunk 3]
```diff
@@ -1034,37 +1030,81 @@
 
             <dependency>
                 <groupId>org.apache.hudi</groupId>
-                <artifactId>hudi-spark3_${scala.binary.version}</artifactId>
+                <artifactId>hudi-spark-common_${scala.binary.version}</artifactId>
```
Comment (yanghua) [inline pom.xml:1033]: +1 for introducing the common artifact

**Comment by yanghua** [comment]
thought: shall we introduce the integration test for the third-party's basic function, e.g. write path & read path?

**Pull request closed without merging**